### DISCUSSION TITLE
Best way to add a custom button?

### DISCUSSION THREAD

**Discussion by bruceharrison1984** [igordanchenko/yet-another-react-lightbox#18] [Q&A] [answered]
I'd like to add a custom button to delete an image. I can see that you have a plugin interface defined, so should I go this route to add a button or is there another approach that you would recommend?

Really great job on this, it's been awesome to work with so far.

**Comment by igordanchenko** [comment, marked as answer]
There's an option to add custom buttons to the toolbar via `toolbar.buttons` prop. But if you want to add a standalone button that you'd position yourself, then plugin is going to be your best option. Take a look at [Captions](https://github.com/igordanchenko/yet-another-react-lightbox/blob/main/src/plugins/captions/Captions.tsx) plugin as a starting point. You can add a node as a child of the controller module and position it absolutely.

**Reply by bruceharrison1984** [reply]
Thanks for the quick answer!
